pkg/e2e/osd/dedicatedadmin.go

```diff
@@ -70,5 +75,135 @@ var _ = ginkgo.Describe(dedicatedAdminTestName, func() {
+
+		// can create edit rolebinding
+		ginkgo.It("can create edit rolebinding", func() {
+
+			daGroup, err := h.User().UserV1().Groups().Get(context.TODO(), "dedicated-admins", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			daGroup.Users = append(daGroup.Users, "dummy-user@redhat.com")
+			daGroup, err = h.User().UserV1().Groups().Update(context.TODO(), daGroup, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "dummy-admin@redhat.com",
+				Groups: []string{
+					"dedicated-admins",
+				},
+			})
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
+			}()
+
+			_, err = createRolebinding("dummy-user@redhat.com", "edit", h)
+			Expect(err).NotTo(HaveOccurred())
+
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// sa can create edit rolebinding
+		ginkgo.It("sa can create edit rolebinding", func() {
+
+			daGroup, err := h.User().UserV1().Groups().Get(context.TODO(), "dedicated-admins", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			daGroup.Users = append(daGroup.Users, "dummy-user@redhat.com")
+			daGroup, err = h.User().UserV1().Groups().Update(context.TODO(), daGroup, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			h.SetServiceAccount("system:serviceaccount:%s:dedicated-admin-project")
+
+			_, err = createRolebinding("dummy-user@redhat.com", "edit", h)
+			Expect(err).NotTo(HaveOccurred())
+
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+
+		// sa can create admin rolebinding
+		ginkgo.It("sa can create admin rolebinding", func() {
+
+			daGroup, err := h.User().UserV1().Groups().Get(context.TODO(), "dedicated-admins", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			daGroup.Users = append(daGroup.Users, "dummy-user@redhat.com")
+			daGroup, err = h.User().UserV1().Groups().Update(context.TODO(), daGroup, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			h.SetServiceAccount("system:serviceaccount:%s:dedicated-admin-project")
+
+			_, err = createRolebinding("dummy-user@redhat.com", "admin", h)
+			Expect(err).NotTo(HaveOccurred())
+
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
 	})
 })
+
+func createNamespace(namespace string, h *helper.H) (*v1.Namespace, error) {
+
+	log.Printf("Creating namespace for dedicated-admin permissions (%s)", namespace)
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+	h.Kube().CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{})
+
+	// Wait for the namespace to create
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		if _, err := h.Kube().CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{}); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+
+	return ns, err
+}
+
+func createRolebinding(user string, clusterrole string, h *helper.H) (*rbacv1.RoleBinding, error) {
```

Comment:
Are you creating a role binding or a clusterrolebinding? Its not clear why you're taking a clusterrole just from the name of the function. As a potential user of this function, I would expect to be able to pass a role or clusterrole to the func

Reply:
The goal is to create a rolebinding. I changed the function so you can pass in a role or clusterrole.